ape_hardhat/providers.py

```diff
@@ -522,7 +522,7 @@ def build_command(self) -> List[str]:
 
         return cmd
 
-    def reset_fork(self):
+    def reset_fork(self, new_block_number: int):
         self._make_request(
-            "hardhat_reset", [{"jsonRpcUrl": self.fork_url, "blockNumber": self.fork_block_number}]
+            "hardhat_reset", [{"forking":{"jsonRpcUrl": self.fork_url, "blockNumber": new_block_number}}]
```

Comment:
it would be nice to include a test in `test_fork_provider.py`:

```python
@pytest.mark.fork
def test_reset_fork_specify_block_number(networks, create_fork_provider):
    provider = create_fork_provider(9020)
    provider.connect()
    provider.mine()
    prev_block_num = provider.get_block("latest").number
    provider.reset_fork(block_number=prev_block_num)
    block_num_after_reset = provider.get_block("latest").number
    assert block_num_after_reset < prev_block_num
    provider.disconnect()

```

Reply:
I changed your suggested code a little bit. In your suggestion the assertion should fail, because after the reset there is the same blocknumber. In my suggestion blocknumber is decreased for one block after the reset.